demos/common/pipelines/src/async_pipeline.cpp

```diff
@@ -62,7 +62,18 @@ AsyncPipeline::AsyncPipeline(std::unique_ptr<ModelBase>&& modelInstance, const C
     execNetwork = engine.LoadNetwork(cnnNetwork, cnnConfig.devices, cnnConfig.execNetworkConfig);
 
     // --------------------------- 5. Create infer requests ------------------------------------------------
-    requestsPool.reset(new RequestsPool(execNetwork, cnnConfig.maxAsyncRequests));
+    unsigned int nireq = cnnConfig.maxAsyncRequests;
+    if (nireq == 0) {
+        try {
+                // +1 to use it as a buffer of the pipeline
+                nireq = execNetwork.GetMetric(METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS)).as<unsigned int>() + 1;
+            } catch (const details::InferenceEngineException& ex) {
+                throw std::runtime_error(std::string("Every device used with the demo should support "
+                    "OPTIMAL_NUMBER_OF_INFER_REQUESTS ExecutableNetwork metric. Failed to query the metric with error: ") + ex.what());
+            }
```

Comment:
Wrong indentation here - should be shifted one level to the left.

Reply:
Shifted